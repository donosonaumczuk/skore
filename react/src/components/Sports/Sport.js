import React from 'react';
import Proptypes from 'prop-types';
import SportInfo from '../match/SportInfo';
import SportPlayers from './SportPalyers';

const getSportImageUrl = links => {
    let sportImageUrl;
    links.forEach(link => {
        if (link.rel === "image") {
            sportImageUrl = link.href;
        }
    });
    return sportImageUrl;
}

const Sport = ({ sport }) => {
    //TODO maybe make image bigger and name bigger, add more staff to sport
    const sportImageUrl = getSportImageUrl(sport.links);
    return (
        <div className="row p-2 mt-2 ml-5 mr-5 match-card rounded-border">
            <div className="col">
                <div className="row mb-4">
                    <SportInfo sportImageUrl={sportImageUrl} sport={sport.displayName} />
                    <SportPlayers playersQuantity={sport.playerQuantity} />
                </div>
            </div>
        </div>
    );
}

Sport.propTypes = {
    sport: Proptypes.object.isRequired//TODO add custom sport proptype
}

export default Sport;