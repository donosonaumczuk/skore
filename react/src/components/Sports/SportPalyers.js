import React from 'react';
import Proptypes from 'prop-types';
import i18next from 'i18next';

const SportPlayers = ({ playersQuantity }) => {
    return (
        <div className="row">
            <div className="col">
                <i className="name-label fas fa-users mr-2"></i>
                <label>{i18next.t('sports.playersPerTeam')}</label>
                 <span>{` ${playersQuantity}`}</span>
            </div>
        </div>
    );
}

SportPlayers.propTypes = {
    playersQuantity: Proptypes.number.isRequired
}

export default SportPlayers;