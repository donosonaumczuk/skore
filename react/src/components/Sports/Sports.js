import React, { Component } from 'react';
import InfiniteScroll from 'react-infinite-scroll-component';
import Sport from './Sport';
import SportService from '../../services/SportService';
import Loader from '../Loader';

const INITIAL_OFFSET = 0;
const QUERY_QUANTITY = 10;
class Sports extends Component {
    mounted = false;
    constructor(props) {
        super(props);
        this.state = {
            sports: [],
            offset: INITIAL_OFFSET,
            total: QUERY_QUANTITY,
            hasMore: true
        }
    }

    addSports = sports => {
        this.setState({
            sports: [...this.state.sports, ...sports],
            offset: this.state.offset + sports.length,
            total: QUERY_QUANTITY,
            hasMore: true
        })
    }

    getSports = async () => {
        const sports = await SportService.getSports();
        if (sports.status) {
            //TODO handle error
        }
        else if (this.mounted) {
            this.addSports(sports);
        }
    }

    componentDidMount = async () => {
        this.mounted = true;
        this.getSports();
    }

    render() {
        if (this.state.sports.length === 0 && this.state.hasMore) {
            return <Loader />;
        }
        else {
            return (
                <div className="match-container container-fluid">
                    {/* TODO sport Filter */}
                    <InfiniteScroll dataLength={this.state.sports.length} next={this.getSports}
                                    hasMore={this.state.hasMore} loader={<Loader />}>
                        {this.state.sports.map((sport) => <Sport key={sport.sportName} sport={sport} />)}
                    </InfiniteScroll>
                </div>
            );
        }

    }

    componentWillUnmount = () => {
        this.mounted = false
    }
}

export default Sports;