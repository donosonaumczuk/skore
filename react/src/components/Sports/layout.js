import React from 'react';
import PropTypes from 'prop-types';
import InfiniteScroll from 'react-infinite-scroll-component';
import Sport from './components/Sport';
import Loader from '../Loader';

const Sports = (props) => {
    const { sports, getSports, hasMore } = props;
    return (
        <div className="match-container container-fluid">
            {/* TODO sport Filter */}
            <InfiniteScroll dataLength={sports.length} next={getSports}
                            hasMore={hasMore} loader={<Loader />}>
                {sports.map((sport) => 
                    <Sport key={sport.sportName} sport={sport} />)}
            </InfiniteScroll>
        </div>
    );
}

Sports.propTypes = {
    sports: PropTypes.array.isRequired,//TODO make sport array custom proptype
    getSports: PropTypes.func.isRequired,
    hasMore: PropTypes.bool.isRequired
}

export default Sports;