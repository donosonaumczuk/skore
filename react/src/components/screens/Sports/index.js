import React, { Component } from 'react';
import SportService from '../../../services/SportService';
import Loader from '../../Loader';
import Utils from './../../utils/Utils';
import Sports from './layout';

const INITIAL_OFFSET = 0;
const QUERY_QUANTITY = 10;

class SportsContainer extends Component {
    mounted = false;
    constructor(props) {
        super(props);
        this.state = {
            sports: [],
            offset: INITIAL_OFFSET,
            total: QUERY_QUANTITY,
            hasMore: true
        }
    }

    addSports = response => {
        const  { sports } = response;
        const hasMore = Utils.hasMorePages(response.links);
        this.setState({
            sports: [...this.state.sports, ...sports],
            offset: this.state.offset + sports.length,
            total: QUERY_QUANTITY,
            hasMore: hasMore
        })
    }

    getSports = async () => {
        const response = await SportService.getSports();
        if (response.status) {
            //TODO handle error
        }
        else if (this.mounted) {
            this.addSports(response);
        }
    }

    componentDidMount = async () => {
        this.mounted = true;
        this.getSports();
    }

    render() {
        if (this.state.sports.length === 0 && this.state.hasMore) {
            return <Loader />;
        }
        else {
            const { sports, hasMore } = this.state;
            return (
                <Sports sports={sports} getSports={this.getSports} hasMore={hasMore} />
            );
        }

    }

    componentWillUnmount = () => {
        this.mounted = false
    }
}

export default SportsContainer;