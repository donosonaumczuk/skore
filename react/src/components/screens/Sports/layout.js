import React from 'react';
import PropTypes from 'prop-types';
import InfiniteScroll from 'react-infinite-scroll-component';
import Sport from './components/Sport';
import Loader from '../../Loader';
import SportPropType from '../../../proptypes/SportPropType';
import WithError from '../../hocs/WithError';
import WithExecuting from '../../hocs/WithExecuting';
import WithLoading from '../../hocs/WithLoading';

const Sports = ({ sports, getSports, hasMore, currentUser, likes,
                    likeSport, dislikeSport }) => {
        return (
        <div className="match-container container-fluid container p-2">
            <InfiniteScroll dataLength={sports.length} style={{height: 'auto', overflow: 'visible'}} next={getSports}
                            hasMore={hasMore} loader={<Loader />}>
                {sports.map((sport) =>
                    <Sport key={sport.sportName} sport={sport}
                            isLogged={currentUser}
                            isLiked={likes[sport.sportName]}
                            likeSport={likeSport}
                            dislikeSport={dislikeSport} />)}
            </InfiniteScroll>
        </div>
    );
}

Sports.propTypes = {
    sports: PropTypes.arrayOf(SportPropType),
    getSports: PropTypes.func.isRequired,
    hasMore: PropTypes.bool.isRequired,
    currentUser: PropTypes.string,
    likes: PropTypes.object.isRequired,
    likeSport: PropTypes.func.isRequired,
    dislikeSport: PropTypes.func.isRequired
}

export default WithError(WithExecuting(WithLoading(Sports)));
