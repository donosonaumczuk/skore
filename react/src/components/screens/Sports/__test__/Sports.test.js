import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Sports from '../layout';

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useHistory: () => ({
      push: jest.fn()
    })
}));


test('Sports snapshot test', () => {
    //set up
    const sports = [
        {
            sportName: "SportName",
            displayName: "Display name",
            playerQuantity: 1,
            links: [
                {
                    rel: "image",
                    "href": "sportImageUrl"
                }
            ]
        }
    ];
    const getSports = () => "sport function";
    const hasMore = false;

    //execution
    const component = renderer.create(
      <Sports sports={sports} getSports={getSports} hasMore={hasMore} />,
    );
    let tree = component.toJSON();

    //postconditions
    expect(tree).toMatchSnapshot();
});

test('enzyme demo', () => {
    const sports = [
        {
            sportName: "SportName",
            displayName: "Display name",
            playerQuantity: 1,
            links: [
                {
                    rel: "image",
                    "href": "sportImageUrl"
                }
            ]
        }
    ];
    const wrapper = shallow(<Sports sports={sports} />);
    expect(wrapper.exists()).toBeTruthy();
})
