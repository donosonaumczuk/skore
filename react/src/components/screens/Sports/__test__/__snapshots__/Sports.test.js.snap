// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sports snapshot test 1`] = `
<div
  className="match-container container-fluid"
>
  <div
    className="infinite-scroll-component__outerdiv"
    style={Object {}}
  >
    <div
      className="infinite-scroll-component "
      style={
        Object {
          "WebkitOverflowScrolling": "touch",
          "height": "auto",
          "overflow": "auto",
        }
      }
    >
      <div
        className="row p-2 mt-2 ml-5 mr-5 match-card rounded-border"
      >
        <div
          className="col"
        >
          <div
            className="row mb-4"
          >
            <div
              className="col-2 col-sm-3"
            >
              <div
                className="container-fluid pt-2"
              >
                <div
                  className="row"
                >
                  <div
                    className="col col-xl-4 mr-0 mt-1"
                  >
                    <img
                      alt="sport-pic"
                      className="sport-img"
                      src="sportImageUrl"
                    />
                  </div>
                  <div
                    className="col-6 col-xl d-none d-sm-block pl-0"
                  >
                    <p
                      className="sport-label"
                    >
                      Display name
                    </p>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="row"
            >
              <div
                className="col"
              >
                <i
                  className="name-label fas fa-users mr-2"
                />
                <label />
                <span>
                   1
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
